apps/openmw/mwworld/actionharvest.cpp

```diff
@@ -0,0 +1,90 @@
+#include "actionharvest.hpp"
+
+#include <sstream>
+
+#include <boost/format.hpp>
+
+#include <MyGUI_LanguageManager.h>
+
+#include "../mwbase/environment.hpp"
+#include "../mwbase/mechanicsmanager.hpp"
+#include "../mwbase/windowmanager.hpp"
+#include "../mwbase/world.hpp"
+
+#include "class.hpp"
+#include "containerstore.hpp"
+
+namespace MWWorld
+{
+    ActionHarvest::ActionHarvest (const MWWorld::Ptr& container)
+        : Action (true, container)
+    {
+        setSound("Item Ingredient Up");
+    }
+
+    void ActionHarvest::executeImp (const MWWorld::Ptr& actor)
+    {
+        if (!MWBase::Environment::get().getWindowManager()->isAllowed(MWGui::GW_Inventory))
+            return;
+
+        MWWorld::Ptr target = getTarget();
+        MWWorld::ContainerStore& store = target.getClass().getContainerStore (target);
+        MWWorld::ContainerStore& actorStore = actor.getClass().getContainerStore(actor);
+        std::map<std::string, int> takenMap;
+        for (MWWorld::ContainerStoreIterator it = store.begin(); it != store.end(); ++it)
+        {
+            if (!it->getClass().showsInInventory(*it))
+                continue;
+
+            int itemCount = it->getRefData().getCount();
+            actorStore.add(*it, itemCount, actor);
+            store.remove(*it, itemCount, getTarget());
+            MWBase::Environment::get().getMechanicsManager()->itemTaken(actor, *it, target, itemCount);
+            takenMap[it->getClass().getName(*it)]+=itemCount;
+        }
+
+        // Spawn a messagebox (only for items added to player's inventory)
+        if (actor == MWBase::Environment::get().getWorld()->getPlayerPtr())
+        {
+            std::ostringstream stream;
+            int lineCount = 0;
+            const static int maxLines = 10;
+            for (auto & pair : takenMap)
+            {
+                std::string itemName = pair.first;
+                int itemCount = pair.second;
+                lineCount++;
+                if (lineCount == maxLines)
+                    stream << "\n...";
+                else if (lineCount > maxLines)
+                    break;
+
+                // The two GMST entries below expand to strings informing the player of what, and how many of it has been added to their inventory
+                std::string msgBox;
+                if (itemCount == 1)
+                {
+                    msgBox = MyGUI::LanguageManager::getInstance().replaceTags("\n#{sNotifyMessage60}");
+                    msgBox = boost::str(boost::format(msgBox) % itemName);
```

Comment:
Use the syntax introduced in [this commit](https://github.com/OpenMW/openmw/commit/8ecd0b82a4604d7bc5e34b663c218c9c005dd34b) instead so the Boost dependency can be avoided.